### PR TITLE
move Get*(v1.Pod) into a new podinfo package

### LINKED ISSUE #71863: [scheduler cleanup phase 2]: Rename `pkg/scheduler/util` to `pkg/scheduler/internal/podinfo`
**PLEASE COORDINATE WITH SIG-SCHEDULING BEFORE WORKING ON THIS ISSUE.** We want to avoid the situation where two people accidentally start work on the same fix.

Main tracking issue: https://github.com/kubernetes/kubernetes/issues/68951

If you contributed in phase 1, please give others a chance before claiming this issue. You can also help out by reviewing the phase 2 PRs.

This package contains only functions related to pods. Having “util” as a name is generally discouraged, since it doesn’t tell readers anything about what is in that package. This change will also mirror the change to create `pkg/scheduler/nodeinfo` which will have a bunch of node-related types and functions.

`GetPodPriority` from this package is used by `pkg/kubelet/eviction`. In order to make our package internal, we should do one of the following. (1 is my preference)

1. Copy the function into kubelet and remove the dependency
2. Move the function into a different area like `apimachinery` or `client-go`
3. Make the funcion an official part of the scheduler’s public go libraries

/sig scheduling
/kind cleanup
/help
/good-first-issue

### PR DESCRIPTION
**Pull request by hprateek43**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
This is in reference to PR #76990 referencing to the issue #71863. The previous PR was large so is being refactored into smaller PRs as per recommendation.
**Which issue(s) this PR fixes**:

Fixes #71863

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by hprateek43** [comment]
@alculquicondor Can you review the changes in reference to PR #76990

**Review by draveness — COMMENTED** [review]
/ok-to-test

File: pkg/api/pod/util.go [hunk 1]
```diff
@@ -842,3 +844,14 @@ func multiplePodIPsInUse(podStatus *api.PodStatus) bool {
 	}
 	return false
 }
+
+// GetPodPriority returns priority of the given pod.
+func GetPodPriority(pod *v1.Pod) int32 {
```
Comment (draveness) [inline pkg/api/pod/util.go]: Why are we moving this function here?

File: pkg/kubelet/eviction/helpers.go [hunk 2]
```diff
@@ -23,15 +23,15 @@ import (
...
+	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	"k8s.io/klog"
 	v1resource "k8s.io/kubernetes/pkg/api/v1/resource"
 	statsapi "k8s.io/kubernetes/pkg/kubelet/apis/stats/v1alpha1"
 	evictionapi "k8s.io/kubernetes/pkg/kubelet/eviction/api"
 	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
-	schedulerutils "k8s.io/kubernetes/pkg/scheduler/util"
 	volumeutils "k8s.io/kubernetes/pkg/volume/util"
+	podinfo "k8s.io/kubernetes/pkg/api/pod"
```
Comment (draveness) [inline pkg/kubelet/eviction/helpers.go]: this is weird and confusing, podutil could be a better same as other packages import this with `podutil`

File: pkg/scheduler/algorithm/predicates/predicates.go [hunk 3]
```diff
@@ -46,9 +46,10 @@ import (
 	priorityutil "k8s.io/kubernetes/pkg/scheduler/algorithm/priorities/util"
 	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"
 	schedulernodeinfo "k8s.io/kubernetes/pkg/scheduler/nodeinfo"
-	schedutil "k8s.io/kubernetes/pkg/scheduler/util"
 	"k8s.io/kubernetes/pkg/scheduler/volumebinder"
 	volumeutil "k8s.io/kubernetes/pkg/volume/util"
+	podinfo "k8s.io/kubernetes/pkg/scheduler/internal/podinfo"
+
```
Comment (draveness) [inline pkg/scheduler/algorithm/predicates/predicates.go]: extra line

File: pkg/scheduler/core/extender_test.go [hunk 4]
```diff
@@ -37,6 +37,7 @@ import (
 	schedulernodeinfo "k8s.io/kubernetes/pkg/scheduler/nodeinfo"
 	schedulertesting "k8s.io/kubernetes/pkg/scheduler/testing"
 	"k8s.io/kubernetes/pkg/scheduler/util"
+	podinfo "k8s.io/kubernetes/pkg/api/pod"
```
Comment (draveness) [inline pkg/scheduler/core/extender_test.go]: ditto

File: pkg/scheduler/core/generic_scheduler.go [hunk 5]
```diff
@@ -36,6 +36,7 @@ import (
 	"k8s.io/apimachinery/pkg/util/errors"
 	corelisters "k8s.io/client-go/listers/core/v1"
 	"k8s.io/client-go/util/workqueue"
+	podinfo "k8s.io/kubernetes/pkg/api/pod"
```
Comment (draveness) [inline pkg/scheduler/core/generic_scheduler.go]: ditto

File: pkg/scheduler/util/heap.go [hunk 6]
```diff
@@ -240,12 +241,12 @@ func (h *Heap) Len() int {
 }
 
 // NewHeap returns a Heap which can be used to queue up items to process.
-func NewHeap(keyFn KeyFunc, lessFn LessFunc) *Heap {
+func NewHeap(keyFn KeyFunc, lessFn podinfo.LessFunc) *Heap {
```
Comment (draveness) [inline pkg/scheduler/util/heap.go]: We should move LessFunc to this file IMO.

File: pkg/api/pod/util.go (hunk 1 above)
Comment (draveness) [reply pkg/api/pod/util.go]: The podinfo package has exactly the same function.

**Review by draveness — CHANGES_REQUESTED** [review]

File: pkg/scheduler/internal/podinfo/podinfo_test.go [hunk 7]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package util
+package podinfo
```
Comment (k82cn) [inline pkg/scheduler/internal/podinfo/podinfo_test.go]: why are we going to move this into internal util pkg? It maybe useful for customized scheduler by framework.

File: pkg/api/pod/util.go (hunk 1 above)
Comment (hprateek43) [reply pkg/api/pod/util.go]: Should I keep it here or in podinfo?

File: pkg/scheduler/internal/podinfo/podinfo_test.go (hunk 7 above)
Comment (hprateek43) [reply pkg/scheduler/internal/podinfo/podinfo_test.go]: PR #76990 has discussion regarding moving this into an internal package

File: pkg/api/pod/util.go (hunk 1 above)
Comment (draveness) [reply pkg/api/pod/util.go]: > Should I keep it here or in podinfo?

We should keep it in podinfo if it works well.

Comment (alculquicondor) [reply pkg/api/pod/util.go]: +1

Comment (alculquicondor) [reply pkg/api/pod/util.go]: wait, this is what the kubelet uses. But, this should actually go to `pkg/api/v1/pod`.

You could consider open a separate PR just for moving this function, as it involves a different set of reviewers.

File: pkg/scheduler/internal/podinfo/podinfo_test.go (hunk 7 above)
Comment (hprateek43) [reply pkg/scheduler/internal/podinfo/podinfo_test.go]: @k82cn Can you confirm if we can move it into an internal package?

File: pkg/scheduler/util/heap.go (hunk 6 above)
Comment (hprateek43) [reply pkg/scheduler/util/heap.go]: @draveness Can you confirm the changes?

File: pkg/api/pod/util.go (hunk 1 above)
Comment (hprateek43) [reply pkg/api/pod/util.go]: Closing this PR and opening separate PR

**Pull request closed without merging**